zdd.py

```diff
@@ -496,24 +500,38 @@ def deploy_marathon_app(args, app):
 
 
 def get_service_port(app):
-    try:
-        return \
-            int(app['container']['docker']['portMappings'][0]['servicePort'])
-    except KeyError:
-        try:
-            return \
-                int(app['portDefinitions'][0]['port'])
-        except KeyError:
-            return int(app['ports'][0])
+    portMappings = get_app_port_mappings(app)
+    if len(portMappings) > 0:
+        servicePort = portMappings[0].get('servicePort')
+        if servicePort:
+            return servicePort
+    portDefinitions = app.get('portDefinitions', [])
+    if len(portDefinitions) > 0:
+        port = ['portDefinitions'][0].get('port')
+        if port:
+            return int(port)
+    ports = app.get('ports', [])
+    if len(ports) > 0:
+        return int(ports[0])
+    raise MissingFieldException("App doesn't contain a service port",
+                                'container.portMappings')
 
 
 def set_service_port(app, servicePort):
-    try:
-        app['container']['docker']['portMappings'][0]['servicePort'] = \
-          int(servicePort)
-    except KeyError:
-        app['ports'][0] = int(servicePort)
-
+    container = app.get('container', {})
+    portMappings = container.get('docker', {}).get('portMappings', [])
+    if len(portMappings) > 0:
+        portMappings[0]['servicePort'] = int(servicePort)
+        app['container']['docker']['portMappings'] = portMappings
+        return app
+    portMappings = container.get('portMappings', [])
+    if len(portMappings) > 0:
+        app['container']['portMappings'][0]['servicePort'] = int(servicePort)
```

Comment:
No I only made the change to stay under 80 characters because of some lint checks. I resolved it another way in next commit.